ml-agents/mlagents/trainers/saver/saver.py

```diff
@@ -0,0 +1,28 @@
+# # Unity ML-Agents Toolkit
+import abc
+
+
+class BaseSaver(abc.ABC):
+    """This class is the base class for the Saver"""
+
+    def __init__(self):
+        """
+        TBA
+        """
+        pass
+
+    @abc.abstractmethod
+    def register(self, module):
+        pass
+
+    @abc.abstractmethod
+    def save_checkpoint(self, brain_name: str, step: int) -> str:
```

Comment:
nit: I'd move the comments from the concrete implementations to here so we don't have to duplicate them for every subclass of Saver